packages/react-hooks/src/useInflation.ts

```diff
@@ -31,57 +24,91 @@ function calcInflationRewardCurve (minInflation: number, stakedFraction: number,
   ));
 }
 
-function calcInflation (api: ApiPromise, totalStaked: BN, totalIssuance: BN, numAuctions: BN): Inflation {
-  const inflationParams = getInflationParams(api);
-  const { auctionAdjust, auctionMax, falloff, maxInflation, minInflation, stakeTarget } = inflationParams;
-  const stakedFraction = totalStaked.isZero() || totalIssuance.isZero()
-    ? 0
-    : totalStaked.mul(BN_MILLION).div(totalIssuance).toNumber() / BN_MILLION.toNumber();
+function calcInflationOnNonAleph (api: ApiPromise, stakedFraction: number, numAuctions: BN): Inflation {
+  const { auctionAdjust, auctionMax, falloff, maxInflation, minInflation, stakeTarget } = getInflationParams(api);
+
   const idealStake = stakeTarget - (Math.min(auctionMax, numAuctions.toNumber()) * auctionAdjust);
   const idealInterest = maxInflation / idealStake;
-  let inflationInPercentage = 0;
 
-  if ('yearlyInflationInTokens' in inflationParams) {
-    inflationInPercentage = 100 * calcInflationUniformEraPayout(totalIssuance, inflationParams.yearlyInflationInTokens);
-  } else {
-    inflationInPercentage = 100 * calcInflationRewardCurve(minInflation, stakedFraction, idealStake, idealInterest, falloff);
-  }
+  const inflationInPercentage = 100 * calcInflationRewardCurve(minInflation, stakedFraction, idealStake, idealInterest, falloff);
 
-  let stakedReturn = stakedFraction
+  const stakedReturn = stakedFraction
     ? (inflationInPercentage / stakedFraction)
     : 0;
 
+  return {
+    inflation: inflationInPercentage,
+    stakedFraction,
+    stakedReturn
+  };
+}
+
+function calcInflationOnAleph (yearlyInflationInPercentage: number, stakedFraction: number) {
+  const baseStakedReturn = stakedFraction !== 0
+    ? (yearlyInflationInPercentage / stakedFraction)
+    : 0;
+
   // Here we multiply stakedReturn by 0.9, as in case of Aleph Zero chain 10% of return goes to treasury
-  if ('yearlyInflationInTokens' in inflationParams) {
-    stakedReturn *= 0.9;
-  }
+  const stakedReturn = baseStakedReturn * 0.9;
 
   return {
-    idealInterest,
-    idealStake,
-    inflation: inflationInPercentage,
+    inflation: yearlyInflationInPercentage,
     stakedFraction,
     stakedReturn
   };
 }
 
+function useYearlyInflation () {
+  const { api } = useApi();
+
+  const [yearlyInflation, setYearlyInflation] = useState<number>();
+
+  const getYearlyInflation = api.call?.alephSessionApi?.yearlyInflation;
+
+  useEffect(() => {
+    getYearlyInflation?.().then((val) => {
+      setYearlyInflation(val.toNumber() / 1_000_000_000);
+    }).catch(console.error);
+  }, [getYearlyInflation]);
```

Comment:
Would that function really ever change? If yes, do we really want it to change (maybe it should be in `useCallback()`?)? If no, than maybe the `api.call?.alephSessionApi?.yearlyInflation` is misleading.